File: courses/LEETCODE In Python 50 Algorithms Coding Interview Questions/10_math_question_count_primes_(easy)/brute.py
"""Count Primes"""


class Solution:
    def countPrimes(self, n):
        if n < 2:
            return 0
        count = 0
        for i in range(2, n):
            isPrime = True
            for j in range(2, i):
                if i % j == 0:
                    isPrime = False
                    break
            if isPrime:
                count += 1

        return count


if __name__ == "__main__":
    print(Solution().countPrimes(1))
    print(Solution().countPrimes(5))
    print(Solution().countPrimes(10))
    print(Solution().countPrimes(17))

File: courses/LEETCODE In Python 50 Algorithms Coding Interview Questions/10_math_question_count_primes_(easy)/brute.js

/*  Given a number n, return an array of prime numbers whose
    size is equal to the number n. 

    Example: 
    0 -> [], 
    1 -> [2], 
    3 -> [2, 3, 5] 
*/

function getPrimes(num) {
    if (num === 0) {
        return []
    }
    if (num === 1) {
        return [2]
    }

    const length = num ** 2
    let count = 0
    let primes = []
    for (let i = 2; i < length; i++) {
        let isPrime = true
        for (let j = 2; j < i; j++) {
            if (i % j === 0) {
                isPrime = false
            }
        }
        if (isPrime) {
            primes.push(i)
            count++
        }
        if (count === num) break
    }
    
    return primes
}
